spec/models/application_spec.rb

```diff
@@ -0,0 +1,34 @@
+require "rails_helper"
+
+RSpec.describe Application, type: :model do
+  describe "relationships" do
+    it { should have_many(:pets) }
```

Comment:
Should be `it { should have_many(:pets).through(:application_pets) }`

You are also missing the association test for `:application_pets`